sprite.go

```diff
@@ -380,40 +471,44 @@ func (p *Sprite) OnTouched__0(onTouched func(obj *Sprite)) {
 	}
 }
 
-func (p *Sprite) OnTouched__1(onTouched func()) {
-	p.OnTouched__0(func(*Sprite) {
+func (p *SpriteImpl) OnTouched__1(onTouched func()) {
+	p.OnTouched__0(func(data interface{}) {
 		onTouched()
 	})
 }
 
-func (p *Sprite) OnTouched__2(name string, onTouched func(obj *Sprite)) {
-	p.OnTouched__0(func(obj *Sprite) {
-		if obj.name == name {
+func (p *SpriteImpl) OnTouched__2(name string, onTouched func(interface{})) {
+	p.OnTouched__0(func(data interface{}) {
+		obj := data.(*SpriteImpl)
```

Comment:
It is better to pass the `Sprite` instead of `*SpriteImpl` to `onTouched`. We do not want the game developer to access `*SpriteImpl`.